### PR TITLE
fix: refactor UpdateBatch method

### PR DESCRIPTION
**Pull request by uzziahlin**
refactor UpdateBatch method and add test

### CODE REVIEW

File: client/orm/db.go [hunk 1]
```diff
@@ -819,63 +819,121 @@ func (d *dbBase) UpdateBatch(ctx context.Context, q dbQuerier, qs *querySet, mi
...
+				_, _ = buf.WriteString(" >> ?")
 			case ColBitLShift:
-				cols = append(cols, col+" = "+col+" << ?")
+				_, _ = buf.WriteString(" << ?")
 			case ColBitXOR:
-				cols = append(cols, col+" = "+col+" ^ ?")
+				_, _ = buf.WriteString(" ^ ?")
 			case ColBitOr:
-				cols = append(cols, col+" = "+col+" | ?")
+				_, _ = buf.WriteString(" | ?")
```
Comment (flycash) [inline client/orm/db.go:928]: please add test cases to cover all these branches

Comment (uzziahlin) [reply client/orm/db.go:928]: Okay.

File: client/orm/db.go [hunk 2]
```diff
@@ -819,63 +819,121 @@ func (d *dbBase) UpdateBatch(ctx context.Context, q dbQuerier, qs *querySet, mi
...
+	}
+
+	query := buf.String()
+
+	d.ins.ReplaceMarks(&query)
+
+	return query
+}
+
+func (d *dbBase) getSetSQL(cols []string, values []interface{}) string {
```
Comment (flycash) [inline client/orm/db.go:881]: refactor this function:
```go
func (d *dbBase) buildSetSQL(bs Buffer, cols []string, values []any) {

}
```
we want to reuse the buffer to reduce the memory allocation.
And you can still keep passing an empty buffer into your unit tests.
After that, rename the T to `owner` and Q to `quote`, T and Q are not good names.

Comment (flycash) [reply client/orm/db.go:881]: if you don't like the name `owner`, you can give it a new name.

Comment (uzziahlin) [reply client/orm/db.go:881]: Okay.

**Pull request merged**